runtime/domain.c

```diff
@@ -1220,6 +1213,20 @@ void caml_interrupt_self(void) {
   interrupt_domain(&domain_self->interruptor);
 }
 
+void caml_reset_young_limit(caml_domain_state * dom_st)
+{
+  CAMLassert(dom_st == Caml_state);
+  /* An interrupt might have been queued in the meanwhile; this
+     achieves the proper synchronisation. */
+  atomic_exchange(&dom_st->young_limit, (uintnat)dom_st->young_start);
+  if (caml_incoming_interrupts_queued()
+      || atomic_load_explicit(&dom_st->requested_external_interrupt,
```

Comment:
I think this needs `caml_check_pending_signals`. Aside from allowing for asynchronous actions follow-ups, the value from this reorganisation should be that signals don't get delayed if they arrive at an inopportune time.

Reply:
If I understand correctly, `caml_update_young_limit` as implemented here will keep young_limit in interrupt-triggering state if a STW request is still pending, or a "requested external interrupt". On the other hand, signals and explicit GC requests are not checked -- even if some are pending, `young_limit` will be reset to `young_start`. I'm not sure I understand the reasoning for behaving differently on these different types of interruptions -- a documentation comment would help.